src/encoded/static/components/regulome_search.js

```diff
@@ -593,26 +593,34 @@ class RegulomeSearch extends React.Component {
     }
 
     updateDimensions() {
-        const screenWidth = this.applicationRef.offsetWidth;
-        let thumbnailWidth = 0;
-        if (screenWidth > 865) {
-            thumbnailWidth = (this.applicationRef.offsetWidth / 3) - 40;
-        } else {
-            thumbnailWidth = this.applicationRef.offsetWidth - 40;
+        // check for applicationRef because otherwise there are errors during resizing process
+        if (this.applicationRef) {
+            const screenWidth = this.applicationRef.offsetWidth;
+            let thumbnailWidth = 0;
+            if (screenWidth > 787) {
+                thumbnailWidth = (this.applicationRef.offsetWidth / 3) - 40;
+            } else if (screenWidth > 483) {
+                thumbnailWidth = (this.applicationRef.offsetWidth / 2) - 40;
+            } else {
+                thumbnailWidth = this.applicationRef.offsetWidth - 40;
+            }
+            this.setState({
+                thumbnailWidth,
+                screenWidth,
+            });
         }
-        this.setState({
-            thumbnailWidth,
-            screenWidth,
-        });
     }
 
     addNewFiles(searchQuery) {
         return new Promise((ok) => {
             requestSearch(searchQuery).then((results) => {
                 const newFiles = results ? results['@graph'] : [];
-                this.setState(prevState => ({
-                    allFiles: [...prevState.allFiles, ...newFiles],
-                }));
+                // only update state if new files have been retrieved
+                if (newFiles) {
```

Comment:
`newFiles` would always be truthy. If you meant `newfiles.length > 0` that would also take care of the case where `results['@graph'].length` is zero.